data.php

```diff
@@ -0,0 +1,32 @@
+<?php
+require 'functions.php';
+$db = connectToDatabase();
+
+
+if (isset($_POST["name"])) {
+    $name_input = cleanInput($_POST["name"]);
+    $address_input = cleanInput($_POST["address"]);
+    if (($_POST['rating'] >= 0) && ($_POST["rating"] <= 10)) {
+        $rating_input = cleanInput($_POST["rating"]);
+    } else {
+        return "Your rating must be a number from 0 to 10";
```

Comment:
This return statement won't actually do anything, it's only possible to return from within a function.

What I'd suggest doing instead is sending another redirect back to the homepage but with some GET data added on containing an error message, kinda like this:

`header("location:pubs_of_sheffield.php?error=Your rating must be a number from 0 to 10");`

You can then check for the presence of an error on the homepage and display it if it's set.

Reply:
yea i thought the return statement was a bit hopeful tbh